test/util/util.go

```diff
@@ -225,6 +226,30 @@ func WaitForInstanceReconciledGeneration(client v1beta1servicecatalog.Servicecat
 	)
 }
 
+// isServiceInstanceReady returns whether the given instance has a ready condition
+// with status true.
+func isServiceInstanceReady(instance *v1beta1.ServiceInstance) bool {
+	for _, cond := range instance.Status.Conditions {
+		if cond.Type == v1beta1.ServiceInstanceConditionReady {
+			return cond.Status == v1beta1.ConditionTrue
+		}
+	}
+
+	return false
+}
+
+// isServiceInstanceFailed returns whether the instance has a failed condition with
+// status true.
+func isServiceInstanceFailed(instance *v1beta1.ServiceInstance) bool {
+	for _, condition := range instance.Status.Conditions {
+		if condition.Type == v1beta1.ServiceInstanceConditionFailed && condition.Status == v1beta1.ConditionTrue {
```

Comment:
I understand that `isServiceInstanceReady` and `isServiceInstanceFailed` are functions copied from elsewhere. But it makes me sad to see them so close together, doing the same thing, and yet differing slightly. Can we have each one call a common `isServiceInstanceConditionTrue(*v1beta1.ServiceInstance, v1beta1.ServiceInstanceCondition)` function? Barring that, can we be consistent about whether we short-circuit when the condition is found or whether we only short-circuit when the condition is found with a True status?